image/templates/helm/stackrox-secured-cluster/templates/_labels.tpl

```diff
@@ -15,7 +15,8 @@
 {{ $forPod := index . 4 }}
 {{ $_ := set $labels "app.kubernetes.io/name" "stackrox" }}
 {{ $_ = set $labels "app.kubernetes.io/managed-by" $.Release.Service }}
-{{ $_ = set $labels "helm.sh/chart" (printf "%s-%s" $.Chart.Name ($.Chart.Version | replace "+" "_")) }}
+{{/* Shorten the helm.sh/chart label to avoid exceeding the Kubernetes 63 character limit for dirty tags (local deployments) */}}
+{{ $_ = set $labels "helm.sh/chart" (printf "%s-%s" $.Chart.Name ($.Chart.Version | replace "+" "_" | replace "dirty" "d")) }}
```

Comment:
Interesting idea to not touch the `tag` target in the top-level Makefile, but instead just modify the Helm chart.

I don't see why that wouldn't work. But, one remark, to quote from https://helm.sh/docs/chart_best_practices/labels/:

> This should be the chart name and version: `{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}.`

What about you leave the concatenation here as it is and instead modify the `.Chart.Version`?

Reply:
I didn't want to change the tag generation in the Makefile, because people are familiar with `-dirty` git suffixes and changing them might create confusion. And there might be some other side effects that I'm not aware of.

Changing `.Chart.Version` would have do be done in `deriveChartVersion()` in `version.go`. Arguably it's a better place for the fix, but then again simply fixing the label like I did is locating the workaround right next to the source of the problem.

Should I do the replace in `deriveChartVersion()` then?

Reply:
Why would it have to be done in `deriveChartVersion()`? Maybe I am overlooking something, but I thought you could just do something like:

```
version: [< required "" (.Versions.ChartVersion | replace "-dirty" "-d") >]
appVersion: [< required "" (.Versions.MainVersion | replace "-dirty" "-d") >
```

in `Chart.yaml`?

Reply:
True, I don't know why I didn't also consider that option.